### DISCUSSION TITLE
uuid in drift files

### DISCUSSION THREAD

**Discussion by albazel2004** [simolus3/drift#1661] [Q&A]
Is there any way to use uuid as default value of a column in drift files?

**Comment by simolus3** [comment]
Drift files don't have an equivalent of `clientDefault` at the moment.

Porting over that feature would be a bit weird because Dart expressions are evaluated in the context of the database file (since we're generating a part file), but that's not obvious from the drift file. It's also not possible to declare Dart variables in drift files to potentially store a `Uuid` generator.

**Reply by albazel2004** [reply]
Thank you for your replay, but I think you already made variables for a custom query, why don't you add a default clause at the end of a column definition, for example:
my_uuid text UUID KEY
and the analyzer generates a proper clientDefault expression on the part file (db.g.dart).

Thank you again.

**Reply by simolus3** [reply]
Having a special `UUID` clause could work, but it would essentially enforce this feature on everyone and we can't add special support for every such case in the generator and the SQL parser.
